agent/command.go

```diff
@@ -156,12 +156,10 @@ func (a *Agent) runCommand(ctx context.Context, tc *taskContext, logger client.L
 		tc.taskConfig.Expansions.Put(key, newVal)
 	}
 
+	tc.setCurrentCommand(cmd)
 	if options.isTaskCommands || options.failPreAndPost {
-		tc.setCurrentCommand(cmd)
 		tc.setCurrentIdleTimeout(cmd)
```

Comment:
Seems like the idle timeout still needs to be set for the post block to mirror the old `else` block.

Reply:
fixed

Reply:
I'm pretty sure we can just `tc.setCurrentIdleTimeout` without any condition. It will default to the currently-configured idle timeout or fall back to the default anyways.